packages/edit-site/src/components/welcome-guide/template.js

```diff
@@ -24,16 +21,24 @@ export default function WelcomeGuideTemplate() {
 			'core/edit-site',
 			'welcomeGuide'
 		);
-		const { isPage } = select( editSiteStore );
-		const { getCurrentPostType } = select( editorStore );
+		const { getCurrentPostType, getEditorSettings } = select( editorStore );
+		const hasBackNavigation =
+			!! getEditorSettings().onNavigateToPreviousEntityRecord;
 		return (
 			isTemplateActive &&
 			! isEditorActive &&
-			isPage() &&
-			getCurrentPostType() === 'wp_template'
+			getCurrentPostType() === 'wp_template' &&
+			hasBackNavigation
 		);
 	}, [] );
 
+	// The visibility conditions change in such a way that it’s tricky to avoid
+	// an unexpected flash of the component. Using state and debouncing writes
+	// to it works around the issue.
+	const [ isVisible, setIsVisible ] = useState();
+	const debouncedSetIsVisible = useDebounce( setIsVisible, 32 );
```

Comment:
Is this effectively a timeout before reading the `useSelect` return value? Can we guarantee that `32` is always going to work? Otherwise it might end up being fragile.


According to the original PR https://github.com/WordPress/gutenberg/pull/52014 this welcome guide appears

> to the user when they open a page in the site editor and when they edit that page's template.

I'm wondering if we need to rethink how this is triggered, e.g, someone on the click events themselves.

cc @noisysocks / @SaxonF

Reply:
It’s like a timeout but a self-adjusting one. The total "timeout" increases by increments of `32` each time the latest call to `debouncedSetIsVisible` was within `32` milliseconds of the last. I can’t see how it would fail to present the Welcome guide as intended. Though maybe it could still flash when not intended if someone’s device is really stuttering.

> I'm wondering if we need to rethink how this is triggered, e.g, someone on the click events themselves.

I had wanted to try that but shied away from it due to thinking it’d likely be a greater number of changes. That was before this hack seemed necessary, so I'd be glad to look into it further.

Reply:
The latest in approach in d23d85cca3f8f640875417d2d759b16df81114b5 is works well in my testing and is much cleaner. For now we can neglect this conversation.